testcases/InstallUbuntu.py

```diff
@@ -140,240 +138,66 @@ def runTest(self):
             rawc.sendline('')
             rawc.sendline('')
         else:
-            # With a "Normal" BMC rather than a simulator,
-            # we need to go and grab things from the network to netboot
-
-            # We also need to work around an Ubuntu/Debian installer bug:
-            # https://bugs.launchpad.net/ubuntu/+source/netcfg/+bug/713385
-            arp = subprocess.check_output(['arp', self.host.hostname()]).split('\n')[1]
-            arp = arp.split()
-            host_mac_addr = arp[2]
-            print "# Found host mac addr %s", host_mac_addr
-            kernel_args = kernel_args + ' netcfg/choose_interface=%s ' % host_mac_addr
-
-            self.system.goto_state(OpSystemState.PETITBOOT_SHELL)
-            self.c.run_command("wget http://%s:%s/ubuntu/vmlinux" % (my_ip, port))
-            self.c.run_command("wget http://%s:%s/ubuntu/initrd.gz" % (my_ip, port))
-            self.c.run_command("kexec -i initrd.gz -c \"%s\" vmlinux -l" % kernel_args)
-            self.c.get_console().send("kexec -e\n")
-
-        self.system.wait_for_kexec()
+            if not self.conf.args.host_mac:
+                arp = subprocess.check_output(['arp', self.host.hostname()]).split('\n')[1]
+                arp = arp.split()
+                host_mac_addr = arp[2]
+                print "# Found host mac addr %s", host_mac_addr
+            else:
+                host_mac_addr = self.conf.args.host_mac
+            kernel_args = kernel_args + ' netcfg/choose_interface=%s BOOTIF=01-%s' % (host_mac_addr, '-'.join(host_mac_addr.split(':')))
+
+            cmd = "[ -f %s ]&& rm -f %s;[ -f %s ] && rm -f %s;true" % (vmlinux,
+                                                                       vmlinux,
+                                                                       initrd,
+                                                                       initrd)
+            self.c.run_command(cmd)
+            self.c.run_command("wget http://%s:%s/%s" % (my_ip, port, vmlinux))
+            self.c.run_command("wget http://%s:%s/%s" % (my_ip, port, initrd))
+            self.c.run_command("kexec -i %s -c \"%s\" %s -l" % (initrd,
+                                                                kernel_args,
+                                                                vmlinux))
+            rawc = self.c.get_console()
+            rawc.sendline("kexec -e")
 
         # Do things
-        rawc = self.c.get_console()
-        rawc.expect('Network autoconfiguration has succeeded',timeout=300)
-        rawc.expect('Loading additional components',timeout=300)
-        rawc.expect('Setting up the clock',timeout=300)
-        rawc.expect('Detecting hardware',timeout=300)
-        rawc.expect('Partitions formatting',timeout=300)
-        rawc.expect('Installing the base system',timeout=300)
+        rawc.expect('Sent SIGKILL to all processes', timeout=60)
+        rawc.expect('Loading additional components', timeout=300)
```

Comment:
it should ideally be working, let me check if anything missing still....

Reply:
Sure, Thanks

Reply:
just tried two times it worked for me, but it fails at this place consistently, looks like we need more wait from petitboot --> host kernel boot in framework..

```
======================================================================
ERROR [1174.858s]: runTest (testcases.InstallUbuntu.InstallUbuntu)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/sath/op-test-framework/testcases/InstallUbuntu.py", line 193, in runTest
    self.system.goto_state(OpSystemState.OS)
  File "/home/sath/op-test-framework/common/OpTestSystem.py", line 145, in goto_state
    self.state = self.stateHandlers[self.state](state)
  File "/home/sath/op-test-framework/common/OpTestSystem.py", line 208, in run_PETITBOOT
    self.wait_for_kexec()
  File "/home/sath/op-test-framework/common/OpTestSystem.py", line 1258, in wait_for_kexec
    console.expect(['Performing kexec','kexec_core: Starting new kernel'], timeout=60)
  File "/home/sath/op-test-framework/common/OPexpect.py", line 71, in expect
    searchwindowsize=searchwindowsize)
  File "/usr/lib/python2.7/site-packages/pexpect/__init__.py", line 1418, in expect
    timeout, searchwindowsize)
  File "/usr/lib/python2.7/site-packages/pexpect/__init__.py", line 1433, in expect_list
    timeout, searchwindowsize)
  File "/usr/lib/python2.7/site-packages/pexpect/__init__.py", line 1535, in expect_loop
    raise TIMEOUT(str(err) + '\n' + str(self))
```

Reply:
Satheesh, this is a petitboot autboot timout problem, we have 60 sec's timeout in `wait_for_kexec`, so use below command once during start of the test when we enter petitboot shell.
```
nvram --update-config petitboot,timeout=20
```
So the timeout issue will go away with this.

Reply:
@pridhiviraj Sure, have updated it now, for future, it would be helpful to have this handled again globally, by reading timeout value and adjust our timeout accordingly  :-)